File: 3-1.ts
class Mutability {
    static newTotalTime(lapTimes: Array<number>): number {
        let newLapTimes = [...lapTimes]
        newLapTimes.shift();
        let sum = 0;
        for (let x of newLapTimes) {
            sum += x;
        }
        return sum;
    }

    static avgTime(lapTimes: Array<number>): number {
        let time = this.newTotalTime(lapTimes);
        let withoutWarmUp: Array<number> = [...lapTimes];
        let laps = withoutWarmUp.length - 1;
        return time / laps;
    }
}

const myLapTimes = [31.0, 20.9, 21.1, 21.3]

console.log(Mutability.avgTime(myLapTimes))

File: 3-1.java
import java.util.List;

class mutability {
    static double totalTime(List<Double> lapTimes) {
        lapTimes.remove(0);
        double sum = 0;
        for (double x : lapTimes) {
            sum += x;
        }
        return sum;
    }

    static double avgTime(List<Double> lapTimes) {
        double time = totalTime(lapTimes);
        int laps = lapTimes.size();
        return time / laps;
    }
}
